pkg/resources/tag_association_acceptance_test.go

```diff
@@ -32,29 +41,108 @@ func TestAcc_TagAssociation(t *testing.T) {
 		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
-		CheckDestroy: nil,
+		CheckDestroy: acc.CheckTagValueEmpty(t),
 		Steps: []resource.TestStep{
 			{
 				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_TagAssociation/basic"),
-				ConfigVariables: m(),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(resourceName, "object_type", "DATABASE"),
+				ConfigVariables: m(tagId, tagValue),
```

Comment:
Why didn't we use model + custom asserts approach here?

Reply:
As stated in the TODO section of the PR description, I'll do this in the next PR.

Reply:
Used the generated builders for the new tests.